Druid/DRCogitCodeGenerator.class.st

```diff
@@ -831,11 +904,13 @@ DRCogitCodeGenerator >> visitIntToFloatConversion: aDRIntToFloat [
 ]
 
 { #category : #visiting }
-DRCogitCodeGenerator >> visitJump: aDRJump [ 
+DRCogitCodeGenerator >> visitJump: aDRJump [
 
-	self
-		generateJump: #Jump:
-		target: aDRJump target
+	aDRJump asBackJumpLoop ifNotNil: [ :loop |
+		loop loopConditional isJITCompileTimeExpression ifTrue: [
+			^ self "JIT compile loop" ] ].
```

Comment:
Strangee

Reply:
Hahaha you don't want to generate the backjump, it is implicit in the `while`.

But maybe we should move all this JITCompile stuff to subclasses...

Reply:
I think what I meant was that `asBackJump` returning nil is strange :)